File: solutions/julia/roman-numerals/2/roman-numerals.jl
function to_roman(number)
    
    letters = (
        1000 => "M", 
        900 => "CM",
        500 => "D",
        400 => "CD",
        100 => "C",
        90 => "XC",
        50 => "L",
        40 => "XL", 
        10 => "X", 
        9 => "IX", 
        5 => "V", 
        4 => "IV", 
        1 => "I"
    )

    result = ""
    
    if number < 1
        error("Number must be 1 or greater!")
    end
    
    for (value, letter) in letters
        while number >= value
            result *= letter
            number -= value
        end
    end
    result
end

File: solutions/javascript/roman-numerals/2/roman-numerals.js
export const toRoman = (arabic) => {
  const NUMERALS = [
    [1000, 'M'],
    [900, 'CM'],
    [500, 'D'],
    [400, 'CD'],
    [100, 'C'],
    [90, 'XC'],
    [50, 'L'],
    [40, 'XL'],
    [10, 'X'],
    [9, 'IX'],
    [5, 'V'],
    [4, 'IV'],
    [1, 'I'],
  ];

  for (const [number, numeral] of NUMERALS) {
    if (arabic >= number) return numeral + toRoman(arabic - number);
  }

  return '';
};
